core/vm/contracts.go

```diff
@@ -621,10 +621,11 @@ func (c *bls12381G1MultiExp) RequiredGas(input []byte) uint64 {
 	}
 	// Lookup discount value for G1 point, scalar value pair length
 	maxDiscountLen := len(params.Bls12381MultiExpDiscountTable)
+	discountIndex := k
 	if k > maxDiscountLen {
-		k = maxDiscountLen
+		discountIndex = maxDiscountLen
```

Comment:
being an 'index', IMO it should be `len - 1`. The term 'index' to me sounds like it should point to the correct index where to take the value, not the index + 1

Reply:
maybe like this instead: 
```
discountIndex := maxDiscountLen-1
if k < maxDiscountLen {
     discountIndex = k-1 // k > 0 checked above
}
discount := params.Bls12381MultiExpDiscountTable[discountIndex]
```